packages/codemods/documentation/howTo.md

```diff
@@ -0,0 +1,119 @@
+# How to Write and Run a CodeMod
+
+Want to learn how to write and run codemods? You're in the right place!
+
+If you want to learn more about the mods themselves, [check out the official README here](../README.md)
+
+## CodeMod Writing Overview
+
+There are two principal ways of writing your own codemods:
+
+- **Config-based codemod**: Probably the easier way to create a codemod, the config-based codemod reads in the contents of the json file `upgrades.json`, and returns a ready-to-run codemod to the user. To learn more about the specifics of the file configurations, [check out the custom types file](../src/codemods/types.ts)
+- **Custom codemod**: For developers who want more flexibility with their codemods, they have the option to create their own mods by wrapping codemod _utilities_ together in a function and passing said function into the `createCodeMod()` function, which will return a ready-to-run codemod. We'll go through an example of this later.
+
+Here are the current features our codemods support:
+
+- **Prop renaming**: Given a component name, a prop to rename and a replacement name, replace the prop name in all components over a codebase!
+- **Renaming/Repathing imports**: Given a search string / regex, an old name/path, and a new name/path, modify all matching import statements to reflect your desired changes!
+
+## Writing Your First CodeMod
+
+Let's say you were responsible for renaming the `toggled` prop in Fluent's `Dropdown` component to `checked`. There are **two** easy ways to do it!
+
+Feel free to try this out on your local branch of fluent! I've set up the files for you :).
+
+### JSON
+
+- To create a codemod using the config file, all you'd need to do is go into the `upgrades.json` file and replace what's there with the following template as defined in `src/codemods/types.ts`:
+
+```jsonld=
+{
+  "name": // Classify this collection of codemods
+  "upgrades": [
+    {
+      "name": // What upgrade are you running?
+      "type": // Currently renameProp | repathImport
+      "version": // Mod version, as a string
+      "options": {
+        "from": {
+          "importName": // Name of component housing prop
+          "toRename": // Prop name to change
+        },
+        "to": {
+          "replacementName": // New prop name
+        }
+      }
+    }
+  ]
+}
+```
+
+- Once you fill out this file, you can then go to [src/codemods/tests/configMod](../src/codemods/tests/configMod/configMod.test.ts) and test your codemod! Invoke the function `createCodeModFromJson()`, which will turn whatever exists in `upgrades.json` into a list of codemods. Check out ##Testing Your Codemods## for more info on how to verify it works!
+
+### Custom CodeMods
+
+- For developers who want a little more flexibility in their mod writing, you can create a codemod manually using codemod _utilities_. Details on these utilities can be found in `src/codemods/utilities`. To rename a prop, we'll need to use two utility functions:
```

Comment:
There's 1 other option, you can make a codemod by creating a const with type Codemod and then exporting that const object as default from the file. It will then get picked up and run automagically.